h/views/accounts.py

```diff
@@ -139,15 +137,24 @@ def logout(self):
 
     def _redirect_if_logged_in(self):
         if self.request.authenticated_userid is not None:
-            raise httpexceptions.HTTPFound(location=self.login_redirect)
+            raise httpexceptions.HTTPFound(location=self._login_redirect())
+
+    def _login_redirect(self):
+        return self.request.params.get('next', _login_redirect_url(self.request))
```

Comment:
Hmm... you're right. That's a change in behavior I missed; sending the user to the replacement for the stream with a specific query rather than just "the stream". I'm dubious about this since it means we'll most likely be sending new users to a blank page after they login for the first time. I'll flag it up in #design